core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -304,7 +304,8 @@ private[spark] object JsonProtocol {
     ("Finish Time" -> taskInfo.finishTime) ~
     ("Failed" -> taskInfo.failed) ~
     ("Killed" -> taskInfo.killed) ~
-    ("Accumulables" -> accumulablesToJson(taskInfo.accumulables))
+    ("Accumulables" -> accumulablesToJson(taskInfo.accumulables)) ~
+    ("Partition ID" -> taskInfo.partitionId)
```

Comment:
nit: Not sure if it is ideal to maintain the order of the UI display like follow Partition ID follows Task Indec?